ml_tools/tests/test_feature_generator.py

```diff
@@ -0,0 +1,149 @@
+import pytest
+
+from ml_tools import FeatureGenerator, Hour, SimpleAggregator, custom_generator, SingleAggregator, Average
+import pandas as pd
+
+
+def test_not_implemented():
+    with pytest.raises(TypeError, match="Can't instantiate abstract class FeatureGenerator with abstract methods "
+                                        "generate_feature"):
+        FeatureGenerator()
+
+
+# def test_feature_generator_init(single_dataset):
+#     df = single_dataset
+#
+#     test_feature = GenericGenerator(df)
+#
+#     assert test_feature._data.equals(df)
+
+
+def test_hour(timestamps):
+    df = timestamps
+    df_comp = df
+
+    for idx, time in enumerate(df.timestamp):
+        df_comp.timestamp[idx] = time.hour
+
+    new_feature = Hour.generate_feature(df)
+
+    assert new_feature.equals(df_comp)
+
+
+def test_hour_columns(timestamps):
```

Comment:
Have you considered taking a datetime and converting it into a variety of features: month, season, day of the week?  I think that's what featuretools does a lot of.  Derivative feature generation.